packages/react-core/src/components/Button/examples/Button.md

```diff
@@ -18,41 +18,71 @@ import { Link } from '@reach/router';
 
 ### Variations
 
+Each button variant has a particular function and should be implemented consistently across products.
+
+
 ```ts file="./ButtonVariations.tsx"
 ```
 
+| Variation | Description|
+| --- | ---|
+| Primary | A primary button is the most prominent button on a page, used for the most important call to action on a page.|
+| Secondary | Secondary buttons are buttons with less visual prominence than primary buttons. Use secondary buttons for general actions on a page, that don’t require as much emphasis as primary button actions. |
+| Danger Secondary | If secondary buttons include destructive actions, use a dangery secondary button to indicate this risk.|
+| Tertiary | Tertiary buttons are one the least visually prominent buttons and are designed to be less striking than a primary or secondary button, while still retaining a classic button format.  |
+| Danger | Danger buttons are the most prominent of all the button options. Use danger buttons for actions a user can take that are potentially destructive or difficult/impossible to undo, like deleting or removing user data. |
+| Warning | Warning buttons should be used when an action may chage an important setting or behavior, but not in a destructive or irreversible way. |
+| Link | Links buttons are labeled buttons with no background or border. Link buttons can be used for actions that may not need as much emphasis, navigate users to another page within the same window, and/or avigate users to an external page. |
+| External Link | ?|
+| Inline Link | Inline text can be used as a link button.|
+| Danger Link | Danger link are text based, but convey the same urgency as danger buttons.|
+
 ### Disabled
 
+Disable buttons when the action cannot be completed. Every button variant can be disabled.
```

Comment:
To go off of this comment, Matt, do you think it'd be worth including *how* the buttons can be disabled, e.g. "Disabled styling can be applied to any button variant with the `isDisabled` prop"? This would also apply to other button examples i.e. "aria disabled", "call to action" etc.

I don't particularly have a strong opinion on it, but one argument for including the mention in the description would be it immediately notifies users how to do it without having to go through the entire code in the example. Argument against could be we do already have the props table lower on the page, and the actual code for the example.